drivers/i2c/i2c_mcux_lpi2c.c

```diff
@@ -202,6 +212,89 @@ static int mcux_lpi2c_transfer(const struct device *dev, struct i2c_msg *msgs,
 	return ret;
 }
 
+#if CONFIG_I2C_MCUX_LPI2C_BUS_RECOVERY
+static void mcux_lpi2c_bitbang_set_scl(void *io_context, int state)
+{
+	const struct mcux_lpi2c_config *config = io_context;
+
+	gpio_pin_set_dt(&config->scl, state);
+}
+
+static void mcux_lpi2c_bitbang_set_sda(void *io_context, int state)
+{
+	const struct mcux_lpi2c_config *config = io_context;
+
+	gpio_pin_set_dt(&config->sda, state);
+}
+
+static int mcux_lpi2c_bitbang_get_sda(void *io_context)
+{
+	const struct mcux_lpi2c_config *config = io_context;
+
+	return (gpio_pin_get_dt(&config->sda) != 0);
+}
+
+static int mcux_lpi2c_recover_bus(const struct device *dev)
+{
+	const struct mcux_lpi2c_config *config = dev->config;
+	struct mcux_lpi2c_data *data = dev->data;
+	struct i2c_bitbang bitbang_ctx;
+	struct i2c_bitbang_io bitbang_io = {
+		.set_scl = mcux_lpi2c_bitbang_set_scl,
+		.set_sda = mcux_lpi2c_bitbang_set_sda,
+		.get_sda = mcux_lpi2c_bitbang_get_sda,
+	};
+	uint32_t bitrate_cfg;
+	int error = 0;
+
+	if (config->scl.port == NULL || !device_is_ready(config->scl.port)) {
+		LOG_ERR("SCL GPIO not available");
+		return -EIO;
+	}
+
+	if (config->sda.port == NULL || !device_is_ready(config->sda.port)) {
+		LOG_ERR("SDA GPIO not available");
+		return -EIO;
```

Comment:
nit: maybe `-ENODEV` here

Reply:
Yeah, except that's not a valid return code for `i2c_recover_bus()`.